src/parser/parser.cpp

```diff
@@ -480,4 +481,13 @@ ColumnList Parser::ParseColumnList(const string &column_list, ParserOptions opti
 	return std::move(info.columns);
 }
 
+LogicalType Parser::ParseLogicalType(const string &type, ParserOptions options) {
+	const auto expr_list = ParseExpressionList("CAST(NULL AS " + type + ")", options);
```

Comment:
You probably need to put the type input in quotes in case a user/extension creates a type that conflicts with a reserved SQL keyword (such as `ANY`) . e.g.

` "CAST(NULL AS \"" + type + "\")"`

Reply:
resolved by using DBConfig::ParseLogicalType